crates/matrix-sdk/src/encryption/mod.rs

```diff
@@ -13,250 +13,20 @@
-//! [device keys] to the server. Those device keys cannot be replaced once they
-//! have been uploaded and tied to a device ID.
-//!
-//! ### Using an access token
-//!
-//! 1. Log in with the password using [`Client::login()`] setting the
-//!    `device_id` argument to `None`.
-//! 2. Store the access token, preferably somewhere secure.
-//! 3. Use [`Client::restore_login()`] the next time the client starts.
-//!
-//! **Note** that the access token is directly connected to a device ID that
-//! lives on a server. If you're skipping step one of this method, remember that
-//! you **can't** use an access token that already has some device keys tied to
-//! the device ID.
-//!
-//! ### Using a password.
-//!
-//! 1. Log in using [`Client::login()`] setting the `device_id` argument to `None`.
-//! 2. Store the `device_id` that was returned in the login response from the
-//! server.
-//! 3. Use [`Client::login()`] the next time the client starts, make sure to
-//! **set** `device_id` this time to the stored `device_id` from the previous
-//! step. This will replace the access token from the previous login call but
-//! won't create a new device.
-//!
-//! **Note** that the default store supports only a single device, logging in
-//! with a different device id (either `None` or a device ID of another client)
-//! is **not** supported using the default store.
-//!
-//! ## Common pitfalls
-//!
-//! | Failure | Cause | Fix |
-//! | ------------------- | ----- | ----------- |
-//! | No messages get encrypted nor decrypted | The `encryption` feature is disabled | [Enable the feature in your `Cargo.toml` file] |
-//! | Messages that were decryptable aren't after a restart | Storage isn't setup to be persistent | Ensure you've activated the persistent storage backend feature, e.g. `sled_state_store` |
-//! | Messages are encrypted but can't be decrypted | The access token that the client is using is tied to another device | Clear storage to create a new device, read the [Restoring a Client] section |
-//! | Messages don't get encrypted but get decrypted | The `m.room.encryption` event is missing | Make sure encryption is [enabled] for the room and the event isn't [filtered] out, otherwise it might be a deserialization bug |
-//!
-//! [Enable the feature in your `Cargo.toml` file]: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#choosing-features
-//! [Megolm]: https://gitlab.matrix.org/matrix-org/olm/blob/master/docs/megolm.md
-//! [`UserIdentity`]: #struct.verification.UserIdentity
-//! [filtered]: crate::config::SyncSettings::filter
-//! [enabled]: crate::room::Joined::enable_encryption
-//! [Restoring a Client]: #restoring-a-client
-//! [spec]: https://spec.matrix.org/unstable/client-server-api/#relationship-between-access-tokens-and-devices
-//! [device keys]: https://spec.matrix.org/unstable/client-server-api/#device-keys
-
+#![doc = include_str!("../docs/encryption.md")]
```

Comment:
Is there a reason why we would do this? We do it for the README files so we don't have to duplicate crate docs.

But those docs are quite specific for this specific crate and module, I don't see them being useful out of this module. Perhaps they could be restructured so they are useful for the crypto crate as well but that's not the case for now.

Reply:
I did this so rustfmt can format this module again. If I just removed that attribute w/o this change, that would have broken the table in these docs. I also find it nice to have many-hundret-line docs in separate files though, it makes browsing the code as well as reading the latest doc state on GitHub (because it's actually rendered as HTML) nicer.

Reply:
not fond of moving docs away from the code (it makes it easier for them to go out of sync), but I accept that's a pretty fair reason to do it in this case.